pkg/types/azure/disk.go

```diff
@@ -0,0 +1,65 @@
+package azure
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	RxDiskEncryptionSetID   = regexp.MustCompile(`(?i)^/subscriptions/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})/resourceGroups/([-a-z0-9_().]{0,89}[-a-z0-9_()])/providers/Microsoft\.Compute/diskEncryptionSets/([-a-z0-9_]{1,80})$`)
+	RxSubscriptionID        = regexp.MustCompile(`^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`)
+	RxResourceGroup         = regexp.MustCompile(`^[-a-z0-9_().]{0,89}[-a-z0-9_()]$`)
+	RxDiskEncryptionSetName = regexp.MustCompile(`^[-a-z0-9_]{1,80}$`)
+)
+
+// DiskEncryptionSet defines the configuration for a disk encryption set.
+type DiskEncryptionSet struct {
+	SubscriptionID string `json:"subscriptionId"`
+	ResourceGroup  string `json:"resourceGroup"`
+	Name           string `json:"name"`
```

Comment:
These need comments for Go Docs and explain. Writing the comments, adding kubebuilder tags, and generating the explain docs should help clear up errors in the govet and explain tests.

Reply:
Fixed.